dipy/reconst/ivim.py

```diff
@@ -126,15 +133,34 @@ def f_D_star_error(params, gtab, signal, S0, D):
     return signal - f_D_star_prediction([f, D_star], gtab, S0, D)
 
 
-class IvimModel(ReconstModel):
-    """Ivim model
+def ivim_model_selector(gtab, fit_method='LM', *args, **kwargs):
+    """
+    Selector function to switch between the 2-stage Levenberg-Marquardt based
+    NLLS fitting method (also containing the linear fit): `LM` and the Variable
+    Projections based fitting method: `VarPro`.
+
+    Parameters
+    ----------
+    fit_method : string, optional
+        The value fit_method can either be 'LM' or 'VarPro'.
+        default : LM
+
     """
 
-    def __init__(self, gtab, split_b_D=400.0, split_b_S0=200., bounds=None,
-                 two_stage=True, tol=1e-15,
-                 x_scale=[1000., 0.1, 0.001, 0.0001],
-                 options={'gtol': 1e-15, 'ftol': 1e-15,
-                          'eps': 1e-15, 'maxiter': 1000}):
+    if fit_method == 'LM':
+        return IvimModelLM(gtab, fit_method='LM', *args, **kwargs)
+
+    elif fit_method == 'VarPro':
+        return IvimModelVP(gtab, fit_method='LM', *args, **kwargs)
+
+
+IvimModel = ivim_model_selector
+
+
+class IvimModelLM(ReconstModel):
+    """Ivim model
+    """
+    def __init__(self, gtab, fit_method='LM', *args, **kwargs):
```

Comment:
What is the goal of `fit_method` in this class?  What is the goal of `*args` in this class? 

Furthermore, I think it will be better to be explicit and put all keywords arguments  instead of `**kwargs`.